src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3409,7 +3420,7 @@ internal override StrongNameKeys StrongNameKeys
                     filterOpt: filterOpt,
                     cancellationToken: cancellationToken);
 
-                if (!hasDeclarationErrors && !CommonCompiler.HasUnsuppressableErrors(methodBodyDiagnosticBag.DiagnosticBag))
+                if (!hasDeclarationErrors && !CommonCompiler.HasUnsuppressableErrors(methodBodyDiagnosticBag.DiagnosticBag) && filterOpt == null)
```

Comment:
>&& filterOpt == null [](http://example.com/codeflow?start=124&length=20)

For clarity, consider adding a comment and an assert that ```moduleBuilder.IsEncDelta``` is true when ```filterOpt``` is not null.